src/caf/toolkit/cost_utils.py

```diff
@@ -78,6 +82,10 @@ def check_df_col_names(self) -> CostDistribution:
         for col_name, col_val in cols.items():
             if col_val not in self.df:
                 err_cols.update({col_name: col_val})
+        cols.update({"weighted_avg_col": getattr(self, "weighted_avg_col")})
+
+        if self.weighted_avg_col not in self.df.columns:
+            self.df[self.weighted_avg_col] = self.df[self.avg_col]
```

Comment:
Following on from previous comment

Very minor change. I've moved the whitespace down a line so the code reads a bit more like:
1. Check that the required columns are given
2. Add in the "weighted_avg_col" if it doesn't already exist.

```suggestion

        cols.update({"weighted_avg_col": getattr(self, "weighted_avg_col")})
        if self.weighted_avg_col not in self.df.columns:
            self.df[self.weighted_avg_col] = self.df[self.avg_col]
```